smartcontract/service/native/auth/auth.go

```diff
@@ -131,83 +142,79 @@ func AssignFuncsToRole(native *native.NativeService) ([]byte, error) {
 	if err := param.Deserialize(rd); err != nil {
 		return nil, fmt.Errorf("deserialize param failed, caused by %v", err)
 	}
+
+	//prepare event msg
+	contract := param.ContractAddr.ToHexString()
+	failState := []interface{}{"assignFuncsToRole", contract, false}
+	sucState := []interface{}{"assignFuncsToRole", contract, true}
+
 	if param.Role == nil {
-		invokeEvent(native, "assignFuncsToRole", false)
+		pushEvent(native, failState)
 		return utils.BYTE_FALSE, nil
 	}
 
 	//check the caller's permission
-	admin, err := getContractAdmin(native, param.ContractAddr)
+	contractAddr, _ := addressToBytes(&param.ContractAddr)
+	admin, err := getContractAdmin(native, contractAddr)
```

Comment:
why define `addressToBytes` function instead of using `getContractAdmin(native, param.ContractAddr[:])` directly?

Reply:
Although they are equivalent, I prefer to use the `addressToBytes` function because it does not depend how the address type is defined.